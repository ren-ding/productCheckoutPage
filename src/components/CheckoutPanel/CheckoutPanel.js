import React, {Component} from 'react';
import PropTypes from 'prop-types';
import CheckoutProdctSegment from '../CheckoutProductSegment';
import CheckoutSummarySegment from '../CheckoutSummarySegment';

export default class CheckoutPanel extends Component {
    render() {
        return (
            <div className='checkout-panel-wrapper'>
                <h2>CheckoutPanel</h2>
                {this.props.checkoutProducts.map(this.renderCheckoutProductComponent)}
                {this.renderCheckoutSummaryComponent()}
            </div>
        );
    }

    renderCheckoutProductComponent = (product, index) => (
        <CheckoutProdctSegment
            key={index}
            product={product}
            removeFromCheckout={this.props.removeFromCheckout}
        />
    );

    renderCheckoutSummaryComponent = () => (
        <CheckoutSummarySegment
            code = {this.props.code}
            total = {this.props.total}
        />
    );
}

CheckoutPanel.propTypes = {
    checkoutProducts: PropTypes.array.isRequired,
    removeFromCheckout: PropTypes.func.isRequired,
    code: PropTypes.string.isRequired,
    total: PropTypes.number.isRequired
};