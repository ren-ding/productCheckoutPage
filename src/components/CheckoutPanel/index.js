import CheckoutPanel from './CheckoutPanel';
export default CheckoutPanel;