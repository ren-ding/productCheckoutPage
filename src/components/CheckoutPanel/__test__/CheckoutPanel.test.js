import React from 'react';
import CheckoutPanel from '../CheckoutPanel';
import renderer from 'react-test-renderer';

describe('CheckoutPanel', ()=> {
    let props;

    beforeEach(()=> {
        props = {
            checkoutProducts: [],
            removeFromCheckout: ()=>{},
            code:'',
            total:399
        }        
    });

    describe('render', () => {
       it('should render to match snapshot', () => {
           const tree = renderer.create(<CheckoutPanel {...props}/>).toJSON();
           expect(tree).toMatchSnapshot();
       }); 
    });
});