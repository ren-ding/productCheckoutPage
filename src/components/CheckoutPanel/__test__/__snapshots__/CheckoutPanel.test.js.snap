// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutPanel render should render to match snapshot 1`] = `
<div
  className="checkout-panel-wrapper"
>
  <h2>
    CheckoutPanel
  </h2>
  <div
    className="checkout-summary-wrapper"
  >
    <span>
      Promotion Code: 
      
        
    </span>
    <span>
      Total Price: 
      399.00
        
    </span>
  </div>
</div>
`;
